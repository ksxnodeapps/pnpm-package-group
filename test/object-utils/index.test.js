'use strict'
const subject = require('../../lib/object-utils')

describe('entriesReducer', () => {
  it('works', () => {
    expect(
      [
        ['abc', 123],
        ['def', 456],
        ['ghi', 789]
      ].reduce(
        subject.entriesReducer,
        {}
      )
    ).toEqual({
      abc: 123,
      def: 456,
      ghi: 789
    })
  })
})

describe('merge function', () => {
  it('works', () => {
    const args = Array.from('abcdef').map(x => ({arr: [x]}))
    expect(subject.merge(...args)).toEqual({arr: Array.from('abcdef')})
  })

  it('stays unchanged', () => {
    expect(subject.merge(
      ...Array
        .from('abcdef')
        .map(x => ({arr: [x], [x]: x.toUpperCase()}))
    )).toMatchSnapshot()
  })
})
