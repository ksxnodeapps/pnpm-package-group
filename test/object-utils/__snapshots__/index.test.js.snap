// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`merge function stays unchanged 1`] = `
Object {
  "a": "A",
  "arr": Array [
    "a",
    "b",
    "c",
    "d",
    "e",
    "f",
  ],
  "b": "B",
  "c": "C",
  "d": "D",
  "e": "E",
  "f": "F",
}
`;
